sway-lib-std/src/contract_id.sw

```diff
@@ -178,31 +178,24 @@ impl AssetId {
         Self { value: result_buffer }
     }
 
-    /// Creates a new AssetId from a ContractId and the zero SubId.
-    ///
-    /// # Arguments
-    ///
-    /// * `contract_id`: [ContractId] - The ContractId of the contract that created the asset.
+    /// Creates a new AssetId with the default SubId for the current contract.
     ///
     /// # Returns
     ///
-    /// * [AssetId] - The AssetId of the asset. Computed by hashing the ContractId and the zero SubId.
+    /// * [AssetId] - The AssetId of the asset. Computed by hashing the ContractId and the default SubId.
     ///
     /// # Examples
     ///
     /// ```sway
-    /// use std::{callframes::contract_id, constants::ZERO_B256};
+    /// use std::{callframes::contract_id, constants::DEFAULT_SUB_ID};
     ///
     /// fn foo() {
-    ///     let contract_id = contract_id();
-    ///     let sub_id = ZERO_B256;
-    ///
-    ///     let asset_id = AssetId::default(contract_id);
-    ///
-    ///     assert(asset_id == AssetId::new(contract_id, sub_id));
+    ///     let asset_id = AssetId::default();
+    ///     assert(asset_id == AssetId::new(contract_id(), DEFAULT_SUB_ID));
     /// }
     /// ```
-    pub fn default(contract_id: ContractId) -> Self {
+    pub fn default() -> Self {
```

Comment:
I think it makes more sense to have a 'this_default' function than to repurpose this function

Reply:
`default` is canonical